pkg/plugins/datasource_plugin.go

```diff
@@ -73,13 +72,6 @@ func (p *DataSourcePlugin) DataQuery(ctx context.Context, dsInfo *models.DataSou
 	return endpoint.Query(ctx, dsInfo, query)
 }
 
-func (p *DataSourcePlugin) onLegacyPluginStart(pluginID string, client *grpcplugin.LegacyClient, logger log.Logger) error {
-	p.legacyClient = client
-	p.logger = logger
```

Comment:
Are these struct fields still in use or can we remove them as well?

Reply:
I think they are